application/forms/RedisConfigForm.php

```diff
@@ -18,11 +26,79 @@ public function init()
 
     public function createElements(array $formData)
     {
+        $this->addElement('checkbox', 'redis_tls', [
+            'label'       => t('Use TLS'),
+            'description' => t('Encrypt connections to Redis via TLS'),
+            'autosubmit'  => true
+        ]);
+
+        $this->addElement('hidden', 'redis_insecure');
+
+        if (isset($formData['redis_tls']) && $formData['redis_tls']) {
+            $this->addElement('textarea', 'redis_ca', [
+                'label'       => t('Redis CA Certificate'),
+                'description' => sprintf(
+                    t('Verify the peer using this PEM-encoded CA certificate ("%s...")'),
+                    '-----BEGIN CERTIFICATE-----'
+                ),
+                'required'    => true,
+                'validators'  => [$this->wrapIplValidator(X509CertValidator::class, 'redis_ca')]
+            ]);
+
+            $this->addElement('textarea', 'redis_cert', [
+                'label'       => t('Client Certificate'),
+                'description' => sprintf(
+                    t('Authenticate using this PEM-encoded client certificate ("%s...")'),
+                    '-----BEGIN CERTIFICATE-----'
+                ),
+                'validators'  => [$this->wrapIplValidator(X509CertValidator::class, 'redis_cert')]
+            ]);
+
+            $this->addElement('textarea', 'redis_key', [
+                'label'       => t('Client Key'),
+                'description' => sprintf(
+                    t('Authenticate using this PEM-encoded private key ("%s...")'),
+                    '-----BEGIN PRIVATE KEY-----'
+                ),
+                'validators'  => [$this->wrapIplValidator(PrivateKeyValidator::class, 'redis_key')]
+            ]);
+        } else {
+            $this->addElement('hidden', 'redis_ca');
+            $this->addElement('hidden', 'redis_cert');
+            $this->addElement('hidden', 'redis_key');
```

Comment:
Why? It works fine for me without them and the error about the required CA is gone as well when enabling TLS.

Reply:
Haven’t you taught me to do exactly like this? (MSAD vs. OpenLDAP) So I could e.g. switch TLS on, enter certs, switch TLS off, switch TLS on and the certs are still there.

Reply:
I see.